### PR TITLE
Add depth element

### LINKED ISSUE #991: add element to get depth of a ragged list
if not given a list, just return `0`. jelly `ŒḊ`

### PR DESCRIPTION
**Pull request by SE-Nitro**
fixes #991

### CODE REVIEW

File: documents/knowledge/elements.yaml [hunk 1]
```diff
@@ -4564,6 +4564,18 @@
...
 
+- element: "Þj"
+  name: Depth
+  description: Depth of ragged list
+  arity: 1
+  overloads:
+      lst: Depth
+  vectorise: false
+  tests:
+      - "[1, 2, 7] : 3"
```
Comment (Steffan153) [inline documents/knowledge/elements.yaml:4575]:
```suggestion
      - "[1, 2, 7] : 1"
      - "3 : 0"
```

Comment (chunkybanana) [reply documents/knowledge/elements.yaml:4575]: ... what no

**Comment by chunkybanana** [comment]
Also the tests are failing

File: vyxal/elements.py [hunk 2]
```diff
@@ -1023,6 +1023,17 @@ def gen():
...
 
+def depth(lhs, ctx):
+    """Element Þj
+    (lst) -> depth of a
+    """
+    get_depth = lambda d: isinstance(d, list) and max(map(get_depth, d)) + 1
+    depth = get_depth(lhs)
+    if not depth:
+        return 0
+    return depth
```
Comment (Steffan153) [inline vyxal/elements.py:1034]: This can be simpified
```suggestion
    return int(get_depth(lhs)
```

File: documents/knowledge/elements.yaml [hunk 3]
```diff
@@ -4564,6 +4564,19 @@
...
+  name: Depth
+  description: Depth of ragged list
+  arity: 1
+  overloads:
+      lst: Depth
+  vectorise: false
+  tests:
+      - "[1, 2, 7] : 1"
+      - "3 : 0"
+      - "[1, [2, 3, [4]], [69]] : 3"
```
Comment (Steffan153) [inline documents/knowledge/elements.yaml:4577]: woops the list is the stack, so these need wrapped
```suggestion
      - "[[1, 2, 7]] : 1"
      - "[3] : 0"
      - "[[1, [2, 3, [4]], [69]]] : 3"
```

**Review by Steffan153 — APPROVED** [review]
Thanks! I fixed things, so the tests work fine now.

**Review by Lyxal — APPROVED** [review]
Lgtm

**Comment by SE-Nitro** [comment]
oh oops

**Pull request merged**